### DISCUSSION TITLE
No 'pyflash' disk shown in windows 11 file system after flashing micropython firmware for stm32 boards

### DISCUSSION THREAD

**Discussion by infinodeTechnologies** [micropython/micropython#9990] [Hardware & Peripherals]
what I have done:
1 download the firmware (.hex) from the micropython website for my stm32 board(stm32l476 discovery）
2 flash the firmware into the mcu using stm32 cube programmer as recommended by micropython
3 reset the board
4 restart PC

current situation:
the COM port can be detected by windows and micropython's COM port is functional by using a serial terminal such as Putty
**but no 'pyflash' disk would appear (even with using the 'zadig' tool and changing the drivers)**

searching similar problems in the forum only results in few leads which hint issues such as usb cables or driver issues, both of which has been tried and failed to resolve the issue.

has this happen to anyone or any solutions? Thanks

**Comment by scruss** [comment]
Is there any code in `boot.py` setting the USB mode? I think it needs to be set to `pyb.usb_mode('VCP+MSC')` to appear as a flash drive.

MSC mode can cause corruption, since it's not really true USB storage.

**Reply by infinodeTechnologies** [reply]
Thanks for the reply, is there a way of checking the content of boot.py through the serial port? As I can see this stage the serial port will connect to sort of a python console ( REPL) , not the file system in the stm32 board. thanks

**Reply by Plaque-fcc** [reply]
mpremote, yes. And one can use Thonny.
Or in REPL, can use Python itself:
```python
with open('/boot.py', 'r') as bootfile:
    print ("".join(bootfile.readlines()))
```

**Comment by jimmo** [comment]
Which USB port are you using? The ST-Link one or the MCU one?

On the ST-Link one you will only get the serial port. You can use mpremote to access the filesystem over the serial port. https://docs.micropython.org/en/latest/reference/mpremote.html
